framework/src/functions/MooseParsedFunction.C

```diff
@@ -43,20 +43,26 @@ template <typename T>
 Real
 MooseParsedFunctionTempl<T>::value(Real t, const Point & p) const
 {
+  if (!_function_ptr)
+    const_cast<MooseParsedFunctionTempl<T> *>(this)->initialSetup();
```

Comment:
I don't think this can fly, or at least I'd want to see all other avenues exhausted before we went with it

Reply:
ok removed. This was indeed the worst, but the intention was pure so I added an assert